lib/floe/workflow/runner.rb

```diff
@@ -5,29 +5,50 @@ class Workflow
     class Runner
       include Logging
 
-      TYPES         = %w[docker podman kubernetes].freeze
       OUTPUT_MARKER = "__FLOE_OUTPUT__\n"
 
       def initialize(_options = {})
       end
 
+      @runners = {}
       class << self
-        attr_writer :docker_runner
+        # backwards compatibility -- use Floe.set_runner instead
+        def docker_runner=(value)
+          set_runner(value)
+        end
+
+        # see Floe.set_runner
+        def set_runner(schema = "docker", name_or_instance = {}, options = {})
+          # default the schema to docker if not specified
+          if name_or_instance.kind_of?(Hash)
+            schema, name_or_instance, options = "docker", schema, name_or_instance
+          end
 
-        def docker_runner
-          @docker_runner ||= Floe::Workflow::Runner::Docker.new
+          @runners[schema] =
+            case name_or_instance
+            when "docker", nil
+              Floe::Workflow::Runner::Docker.new(options)
+            when "podman"
+              Floe::Workflow::Runner::Podman.new(options)
+            when "kubernetes"
+              Floe::Workflow::Runner::Kubernetes.new(options)
+            when Floe::Workflow::Runner
+              name_or_instance
+            else
+              raise ArgumentError, "docker runner must be one of: docker, podman, kubernetes"
+            end
+        end
+
+        def docker_runner(scheme)
+          set_runner if @runners.empty?
```

Comment:
Shouldn't the scheme be passed along?

```suggestion
        def docker_runner(scheme)
          set_runner(schema) if @runners.empty?
```

Reply:
This sets the default runner.